picoCTF-shell/hacksport/deploy.py

```diff
@@ -70,7 +75,8 @@ def give_port():
             "All usable ports are taken. Cannot deploy any more instances.")
 
     while True:
```

Comment:
Also it doesn't seem like we need a `while True` loop anymore we make `banned_ports` are merged with `unused_ports`, then and `random.choice` ensures we don't have a collision. Not your code but I don't quite understand why #L82 #L83 are checking if the port is already used for this instance+problem and proceeding if it does...